### DISCUSSION TITLE
Unit comparisons?

### DISCUSSION THREAD

**Discussion by quantumtrope** [hgrecco/pint#1692] [Q&A] [answered]
I'm having trouble understanding what is compared between two quantities -- seems like only magnitudes?

```python
import pint
ureg1 = pint.UnitRegistry()
ureg2 = pint.UnitRegistry()

ureg1 == ureg2
# >> False

q1 = ureg1.Quantity(0,"W")
q2 = ureg2.Quantity(0,"W")

q1 == q2
# >> True

# Still `True` if q1 and q2 have a non-zero value:
q1 = ureg1.Quantity(1,"W")
q2 = ureg2.Quantity(1,"W")

q1 == q2
# >> True

ureg1.Quantity == ureg2.Quantity
# >> False

q2.units == q1.units
# >> ValueError: Cannot operate with Unit and Unit of different registries.

q1.magnitude == q2.magnitude
# >> True
```
Is that meant to be the case? I would have expected that comparing two quantities from different registries would throw an error because their units don't match. What am I missing here?

**Comment by hgrecco** [comment, marked as answer]
Comparing units from different registries is not well-defined, and therefore it is an invalid operation. You should not expect a valid result. We rely on Python's consenting adults philosophy.

If you are interested in the details, to avoid a performance hit `Quantity._eq__` compares `q2._units == q1._units`, not `q2.units == q1.units`

**Reply by quantumtrope** [reply]
Good reminder of pythons consenting adults philosophy here. I *knew* that it should fail, but I tried it anyway and was surprised by the result. The detail makes sense and I appreciate the peak under the curtain; I was trying to look through the source code but got lost in the breadcrumb trail.
